content/blog/callpackage-a-tool-for-the-lazy.md

```diff
@@ -0,0 +1,219 @@
+---
+title: callPackage, a tool for the lazy
+date: 2022-08-25
+extra:
+  author: Norbert Melzer
+---
+
+In [`nixpkgs`](https://github.com/nixos/nixpkgs), there is a massive use of the
+`callPackage` function, which provides us with a lot of benefits.
+
+## Basic examples
+
+Though before even discussing the benefits, lets see how it actually gets used.
+
+Given are these 2 files `hello.nix` and `default.nix`:
+
+```nix
+# Usually I would prefer the usage of nix flakes, though that would
+# introduce a lot of boilerplate in the examples for no reason,
+# therefore I will use diamond paths and `import` for simplicity.
+```
```

Comment:
Is that supposed to be a side note? Why not

```suggestion
> **Note**
> Usually I would prefer the usage of Nix flakes.
> But that would introduce a lot of boilerplate in the examples for no reason.
> Therefore I will use diamond paths and `import` for simplicity.
```

otherwise

```suggestion
```nix
# Usually I would prefer the usage of Nix flakes. But that would
# introduce a lot of boilerplate in the examples for no reason.
# Therefore I will use diamond paths and `import` for simplicity.
```
```

Reply:
I would really like to see you qualify why you prefer flakes. I know for a fact that most beginners are utterly confused about flakes (what, why, benefits, support status, ...). So while I suppose you're addressing readers of your own knowledge level, who may object to your choice of example, you're the exception, not the norm. Of the [~21 million professional software developers in the world](https://en.m.wikipedia.org/wiki/Software_engineering_demographics) maybe 50-100k even heard Nix, and something on the order of 2k may be proficient enough to remotely know what you're talking about. That is, at best we'd be missing missing 99,5%  of our target audience if we don't take care of a proper introduction.

Reply:
1. Using a blockquote for the note felt even more wrong semantically :D But yeah, we have to live with what we get in markdown :D
2. I understand your objections, and I think, that it will be the best then, to just remove that note then. Explaining why to use flakes would probably fill another blogpost :D

Reply:
I see you already chaged it, I think we can resolve this then. Thanks :+1:

Reply:
No, that was the other thing with string printing.

Reply:
I'm not sure what GH did here, though I am pretty sure I did the comment on the correct thread…

Reply:
Removed in 1d65f8b381e3e1ff397288ddd6b0a3b653857f5a